src/main/java/org/jboss/forge/arquillian/command/algeron/AlgeronConsumerCreateContractFragmentCommand.java

```diff
@@ -0,0 +1,158 @@
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.arquillian.api.algeron.AlgeronSetupFacet;
+import org.jboss.forge.arquillian.container.model.ContractConsumerLibrary;
+import org.jboss.forge.arquillian.testframework.algeron.AlgeronConsumer;
+import org.jboss.forge.roaster.model.JavaType;
+import org.jboss.forge.roaster.model.source.JavaClassSource;
+
+import javax.inject.Inject;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class AlgeronConsumerCreateContractFragmentCommand extends AbstractProjectCommand implements UICommand  {
+
+   @Inject
+   private ProjectFactory projectFactory;
+
+   @Inject
+   @WithAttributes(shortName = 'j', label = "Test Class", required = true)
+   private UISelectOne<JavaClassSource> testClass;
+
+   @Inject
+   @WithAttributes(shortName = 'c', label = "Consumer Name", required = true)
+   private UIInput<String> consumer;
+
+   @Inject
+   @WithAttributes(shortName = 'p', label = "Provider Name", required = true)
+   private UIInput<String> provider;
+
+   @Inject
+   @WithAttributes(shortName = 'f', label = "Fragment Name", required = true)
+   private UIInput<String> fragment;
+
+   @Inject
+   @WithAttributes(shortName = 't', label = "Contract Type", required = false, enabled = false)
+   private UISelectOne<ContractConsumerLibrary> contractType;
+
+   @Override
+   protected boolean isProjectRequired() {
+      return true;
+   }
+
+   @Override
+   protected ProjectFactory getProjectFactory() {
+      return projectFactory;
+   }
+
+   @Override
+   public UICommandMetadata getMetadata(UIContext context) {
+      return Metadata.from(super.getMetadata(context), getClass())
+              .category(Categories.create("Algeron"))
+              .name("Arquillian Algeron: Create Contract Fragment")
+              .description("This command creates skeleton for contract fragment on given test");
+   }
+
+   @Override
+   public void initializeUI(UIBuilder builder) throws Exception {
+      builder.add(contractType)
+              .add(testClass)
+              .add(consumer)
+              .add(provider)
+              .add(fragment);
+
+      final Project project = getSelectedProject(builder);
+
+      final List<JavaClassSource> sources = new ArrayList<>();
+      project.getFacet(JavaSourceFacet.class).visitJavaTestSources(new JavaResourceVisitor() {
+         @Override
+         public void visit(VisitContext context, JavaResource javaResource) {
+            JavaType<?> javaType;
+            try
+            {
+               javaType = javaResource.getJavaType();
+               if (javaType.isClass()) {
+                  sources.add((JavaClassSource) javaType);
+               }
+            }
+            catch (FileNotFoundException e)
+            {
+               // Do nothing
```

Comment:
Yes it is not necessary to do nothing, since probably it is not going to happen. Notice that this approach is the same used in other Forge Commands. Since it is a visitor pattern the java file is there for sure.